marklogic-client-api/src/main/java/com/marklogic/client/dataservices/impl/InputEndpointImpl.java

```diff
@@ -56,4 +59,35 @@ public BulkInputCaller bulkCaller() {
 		// TODO Auto-generated method stub
 		return null;
 	}
+	
+	final static class BulkInputCallerImpl extends IOEndpointImpl.BulkIOEndpointCallerImpl implements InputEndpoint.BulkInputCaller {
+		
+		private InputEndpointImpl endpoint;
+		private int batchSize;
+		private LinkedBlockingQueue<InputStream> queue;
+		
+		private BulkInputCallerImpl(InputEndpointImpl endpoint, int batchSize) {
+			super(endpoint);
+            this.endpoint = endpoint;
```

Comment:
Please regularize the indentation so the code is easy to read for future maintainers.